transformer_engine/pytorch/csrc/extensions/activation.cpp

```
@@ -24,6 +25,20 @@ py::object activation_helper(const at::Tensor& input, py::handle quantizer, int
   auto [te_output, out] =
       my_quantizer->create_tensor(input_shape, GetTransformerEngineDType(fake_tensor_type));
 
+  // for current scaling, we need to compute amax first and then quantize
+  // because cache cannot fit in the entire tensor to compute amax and quantize
+  // the quantizer should not need amax reduction, no process group needed here
+  if (nvte_tensor_scaling_mode(te_output.data()) == NVTE_CURRENT_TENSOR_SCALING) {
+    // my_quantizer here has to be a Float8CurrentScalingQuantizer
+    auto my_quantizer_cs = static_cast<Float8CurrentScalingQuantizer*>(my_quantizer.get());
+    nvte_compute_amax(te_input.data(), te_output.data(), at::cuda::getCurrentCUDAStream());
+    if (my_quantizer_cs->with_amax_reduction) {
+      NVTE_ERROR(
```

Why is it not supported? Just because of the way it is currently used in Transformer? If there are no blockers then this support should be included for completeness (i assume this would look exactly the same as the allreduce in regular cast).

---

There is no use case for doing an amax reduction in an activation fusion function right now. I personally prefer to raise an error if anyone accidentally called it instead of doing amax reduction under the hood.